### PR TITLE
[cpp-ipc] Add new port 

### PR DESCRIPTION
**Pull request by winsoft666**
**Describe the pull request**

- #### What does your PR fix?  
Add new library named libipc

- #### Which triplets are supported/not supported? Have you updated the [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt)?  
  all, Yes

- #### Does your PR follow the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)?  
 Yes, My PR follow the guide.

- #### If you have added/updated a port: Have you run `./vcpkg x-add-version --all` and committed the result?  
  Yes

### CODE REVIEW

**Comment by JonLiu1993** [comment]
@winsoft666 ,Could you please take a look?
https://dev.azure.com/vcpkg/public/_build/results?buildId=57204&view=artifacts&pathAsName=false&type=publishedArtifacts

**Comment by winsoft666** [comment]
> @winsoft666 ,Could you please take a look?
> https://dev.azure.com/vcpkg/public/_build/results?buildId=57204&view=artifacts&pathAsName=false&type=publishedArtifacts

OK, it is being process

**Comment by winsoft666** [comment]
> _This is a new experimental fast check for PR issues. Please let us know if this bot is helpful!_
> 
> PRs must add only one version and must not modify any published versions

Don't need a new version, this is a new port.

**Comment by JonLiu1993** [comment]
@winsoft666 ,Could you please take a look:
````
CMake Error at scripts/cmake/vcpkg_fail_port_install.cmake:96 (message):
  Target 'OSX' not supported by libipc!
````

File: ports/libipc/CONTROL [hunk 1]
```diff
@@ -0,0 +1,4 @@
+Source: libipc
+Version: 1.0
+Description: C++ IPC Library: A high-performance inter-process communication using shared memory on Linux/Windows.
```
Comment (JonLiu1993) [inline ports/libipc/CONTROL:3]:
```suggestion
Description: C++ IPC Library: A high-performance inter-process communication using shared memory on Linux/Windows.
Supports: !osx
```

**Review by JonLiu1993 — CHANGES_REQUESTED** [review]

**Review by JonLiu1993 — APPROVED** [review]

File: ports/libipc/portfile.cmake [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+    SHA512 de319d48ac396f567b2f5faae87a264b9ebfbaf0559a3e00de61facd6a3f2de1739e064c6e037ad694111ed211c0fb9ea6ec2db0259c100e2bafe553c15ed2f6
+    HEAD_REF master
+)
+
+string(COMPARE EQUAL "${VCPKG_LIBRARY_LINKAGE}" "dynamic" LIBIPC_BUILD_SHARED_LIBS)
+string(COMPARE EQUAL "${VCPKG_CRT_LINKAGE}" "static" LIBIPC_USE_STATIC_CRT)
+
+vcpkg_configure_cmake(
+    SOURCE_PATH ${SOURCE_PATH}
+	PREFER_NINJA
```
Comment (BillyONeal) [inline ports/libipc/portfile.cmake:16]:
```suggestion
    PREFER_NINJA

```

This line has funny indenting for some reason.

File: ports/libipc/CONTROL [hunk 3]
```diff
@@ -0,0 +1,5 @@
+Source: libipc
```
Comment (BillyONeal) [inline ports/libipc/CONTROL:1]: I'm a little concerned to take this name as the name "libipc" is used by a few libraries, such as https://github.com/mheily/libipc and https://github.com/kralf/libipc , which are likely to create customer confusion. Would you find either `mutouyun-libipc` or `cpp-ipc` acceptable names?

File: ports/libipc/portfile.cmake [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+    vcpkg_fixup_cmake_targets(CONFIG_PATH lib/cmake/cpp-ipc)
+elseif(EXISTS ${CURRENT_PACKAGES_DIR}/share/cpp-ipc)
+    vcpkg_fixup_cmake_targets(CONFIG_PATH share/cpp-ipc)
+endif()
+
+file(READ ${CURRENT_PACKAGES_DIR}/include/libipc/export.h LIBIPC_EXPORT_H)
+if(VCPKG_LIBRARY_LINKAGE STREQUAL "dynamic")
+    string(REPLACE "#elif defined(LIBIPC_LIBRARY_SHARED_USING__)" "#elif 1" LIBIPC_EXPORT_H "${LIBIPC_EXPORT_H}")
+endif()
+file(WRITE ${CURRENT_PACKAGES_DIR}/include/libipc/export.h "${LIBIPC_EXPORT_H}")
```
Comment (BillyONeal) [inline ports/libipc/portfile.cmake:36]:
```suggestion
if(VCPKG_LIBRARY_LINKAGE STREQUAL "dynamic")
    vcpkg_replace_string("${CURRENT_PACKAGES_DIR}/include/libipc/export.h" "#elif defined(LIBIPC_LIBRARY_SHARED_USING__)" "#elif 1")
endif()

```

**Review by BillyONeal — CHANGES_REQUESTED** [review]
I think we need to resolve the name question before this is merged, the other comments are nitpicks.

File: ports/libipc/CONTROL (hunk 3 above)
Comment (winsoft666) [reply ports/libipc/CONTROL:1]: I prefer  to use `cpp-ipc`

Comment (BillyONeal) [reply ports/libipc/CONTROL:1]: Please change the name of the port to that then.

**Review by JonLiu1993 — APPROVED** [review]

**Comment by ras0219-msft** [comment]
LGTM, thanks for the PR!

**Pull request merged**